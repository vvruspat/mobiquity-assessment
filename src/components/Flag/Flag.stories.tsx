import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Flag } from "./Flag";
import "../../index.css";

export default {
  title: "UI Kit/Flag",
  component: Flag,
} as ComponentMeta<typeof Flag>;

const Template: ComponentStory<typeof Flag> = (args) => <Flag {...args} />;

export const FlagTemplate = Template.bind({});

FlagTemplate.args = {};
