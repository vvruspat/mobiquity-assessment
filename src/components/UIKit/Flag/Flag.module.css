.flag {
    background-position: center center;
    background-repeat: no-repeat;
    background-size: cover;
    background-color: var(--background-dark-color);
}

.flag_s {
    border-radius: 12px;
    height: 24px;
    width: 24px;
}

.flag_m {
    border-radius: 20px;
    height: 40px;
    width: 40px;
}

.flag_l {
    border-radius: 24px;
    height: 48px;
    width: 48px;
}
