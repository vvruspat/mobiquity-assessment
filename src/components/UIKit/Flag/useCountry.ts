import { useCallback } from "react";
import countries from "flag-icons/country.json";
import nationalities from "./nationality-to-country.json";

export const useCountry = () => {
  const getCountryByName = useCallback((name: string) => {
    return countries.find((country) => country.name === name);
  }, []);

  const getCountryByNationality = useCallback(
    (name: string) => {
      const nationalityCountry = nationalities.find(
        (nationality) => nationality.Nationality === name
      );

      if (nationalityCountry) {
        return getCountryByName(nationalityCountry.Country);
      }
    },
    [getCountryByName]
  );

  return {
    getCountryByName,
    getCountryByNationality,
  };
};
