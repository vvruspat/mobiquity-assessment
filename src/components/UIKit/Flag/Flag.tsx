import { FC, useMemo } from "react";
import cn from "clsx";

import "flag-icons/css/flag-icons.min.css";
import s from "./Flag.module.css";
import { useCountry } from "./useCountry";

type FlagProps = {
  /** Flag */
  size?: "s" | "m" | "l";
  /** Country name from API */
  country?: string;
  /** Nationality name from API */
  nationality?: string;
};

/**
 * Flag component
 */
export const Flag: FC<FlagProps> = ({ size = "s", country, nationality }) => {
  const { getCountryByNationality, getCountryByName } = useCountry();

  const code = useMemo(() => {
    if (nationality) return getCountryByNationality(nationality)?.code;
    if (country) return getCountryByName(country)?.code;
  }, [country, getCountryByName, getCountryByNationality, nationality]);

  return (
    <div
      className={cn(s.flag, s[`flag_${size}`], "fi", `fi-${code}`)}
      data-testid="flag"
    ></div>
  );
};
