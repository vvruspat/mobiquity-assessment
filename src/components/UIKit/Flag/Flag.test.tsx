import { render, screen } from "@testing-library/react";
import { Flag } from "./Flag";

test("renders flag by nationality", () => {
  render(<Flag nationality="Dutch" />);

  const el = screen.getByTestId("flag");

  expect(el).toHaveClass("fi-nl");
  expect(el).toMatchSnapshot();
});

test("renders flag by country", () => {
  render(<Flag country="Netherlands" />);

  const el = screen.getByTestId("flag");

  expect(el).toHaveClass("fi-nl");
  expect(el).toMatchSnapshot();
});
