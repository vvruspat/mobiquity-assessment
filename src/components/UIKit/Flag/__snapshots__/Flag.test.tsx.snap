// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders flag by country 1`] = `
<div
  class="flag flag_s fi fi-nl"
  data-testid="flag"
/>
`;

exports[`renders flag by nationality 1`] = `
<div
  class="flag flag_s fi fi-nl"
  data-testid="flag"
/>
`;
